public/pages/CreateDetector/containers/CreateDetector.tsx

```diff
@@ -61,6 +57,11 @@ export default class CreateDetector extends Component<CreateDetectorProps, Creat
 
   constructor(props: CreateDetectorProps) {
     super(props);
+
+    let detectorState = {}; // if there is detector state in history, then use it to populate all the fields
+    let historyState = this.props.history.location.state as any;
```

Comment:
Minor comment Looks like `historyState` doesn't get reassigned anywhere, right? If not, it could be changed to `const` if there are other revisions planned for this PR.

Reply:
Good catch. Updated.